docs/src/Contributing.md

```diff
@@ -70,7 +70,35 @@ in `docs/`.
 When your PR is ready for review, clean up your commit history by squashing
 and make sure your code is current with `ClimateMachine` master by rebasing.
 
-### Squash and rebase
+## Reformatting your code with `.dev/climaformat.jl`
+
+Your PR cannot be merged unless it conforms to the format style specified
+by the formatter implemented in the script `.dev/climaformat.jl`.
+
+To use the script `.dev/climaformat.jl` to reformat every file in
+`ClimateMachine.jl/` run
+
+```
+julia --project .dev/climaformat.jl
+```
+
+from `ClimateMachine.jl`'s root directory. To format individual files, type
+
+```
+julia --project .dev/climaformat.jl src/individual_file.jl
+```
+
+instead.
+
+!!! warn "Reformatting edits your code"
+    If you would like to be able to revert changes made by the formatter,
+    1. Create a commit that contains your unformatted contributions before running
+       the formatter.
+    2. Reformat your code by running `julia --project .dev/climaformat.jl`.
+    3. Use `git diff` to view changes made by the formatter, and `git stash`
```

Comment:
You can just `git checkout :/`, which will reset it to the state of the index.

Reply:
Ok, I updated the recommendation, let me know you like.

Reply:
My 2 cents: Leave it vaguer and direct users to the git docs page. e.g., "Reformatting may make changes which should be committed [and squashed/rebased](https://clima.github.io/ClimateMachine.jl/latest/Contributing/#Squash-and-rebase)".

Reply:
Isn't it friendly to provide recipes / explicit information? I usually find link-based explanations frustrating and energy-consuming, but maybe I'm alone.